### ISSUE TITLE
Deriving from enum in different file causes spurious warnings

### ISSUE THREAD

**Issue by merc1031** [python/mypy#12398]
Working
`test.py`
```
import enum
from typing import Tuple


class BaseEnum(enum.Enum):
    pass


class MyEnum(BaseEnum):
    A = "A"
    B = "B"
    C = "C"


def use_enum(str_: str) -> Tuple[MyEnum, str]:
    e = MyEnum.A
    if str_ in sorted(x.value for x in MyEnum):
        e = MyEnum(str_)
    return (e, str_)
```

Not working
`test.py`
```
from typing import Tuple

import base


class MyEnum(base.BaseEnum):
    A = "A"
    B = "B"
    C = "C"

def use_enum(str_: str) -> Tuple[MyEnum, str]:
    e = MyEnum.A
    if str_ in sorted(x.value for x in MyEnum):
        e = MyEnum(str_)
    return (e, str_)
```
`base.py`
```
import enum


class BaseEnum(enum.Enum):
    pass
```

errors
```
"Type[MyEnum]" has no attribute "__iter__" (not iterable)  [attr-defined]
Incompatible return value type (got "Tuple[str, str]", expected "Tuple[MyEnum, str]")  [return-value]
```

- Mypy version used: ```0.812```
- Mypy command-line flags: 
- Mypy configuration options from `mypy.ini` (and other config files): 
```
ignore_missing_imports = true
show_error_codes = true
```
- Python version used:
```3.7```
- Operating system and version:
```linux ubuntu 18.04```

**Comment by JelleZijlstra** [comment]
mypy 0.812 is quite old. Can you reproduce with the most recent release?

**Comment by merc1031** [comment]
hmm on further observation, it appears i hadnt saved the new directory structure. The above example does work.
This appears to be caused by a missing `__init__.py` and having the base enum in a directory (a case i was toying with before simplifying for the example.)

New failing example

Not working
`test.py`
```
from typing import Tuple

from enum import base


class MyEnum(base.BaseEnum):
    A = "A"
    B = "B"
    C = "C"

def use_enum(str_: str) -> Tuple[MyEnum, str]:
    e = MyEnum.A
    if str_ in sorted(x.value for x in MyEnum):
        e = MyEnum(str_)
    return (e, str_)
```
`enum/base.py`
```
import enum


class BaseEnum(enum.Enum):
    pass
```
dir structure
```
test.py
enum/
    base.py
```

errors
```
"Type[MyEnum]" has no attribute "__iter__" (not iterable)  [attr-defined]
Incompatible return value type (got "Tuple[str, str]", expected "Tuple[MyEnum, str]")  [return-value]
```
It is fixed by adding the `__init__.py` to the `enum` folder.

This may just be due to the module not being detected due to missing `__init__.py` is this expected? If so feel free to close
